Document human-readable values for Change tracking in Java (#869)
cap-js/docs: java/change-tracking.md

You should consider this when you decide what to include in the identifier.
:::

:::danger Sensitive Data

> Do not use sensitive data as the identifiers.

Author: I added more precise description.

> :::

Reviewer: The same note is in the previous section.
Is there a reason to repeat it?
If so, why is it a danger note this time?

Author: We ignore them anyway. So the previous warning holds true here as well. Removed.

### Identifiers for Associated Entities

When your entity has an association to other entity, you might want to log the changes in the relationships between them.

Given the order entity that has an association to a Customer instead of the element with customer name:
```cds
entity Orders {
  key ID: UUID;
  customer: Association to Customer;
  [...]
}
```

If you annotate such association with `@changelog`, the change log will store the value of the associated entity key, by default.
If you want, you can store some human-readable identifier instead. You define this by annotating the association with own identifier:

```cds
annotate Orders {
  customer @changelog: [ customer.name ]
}
```

Elements from the `@changelog` annotation value must always be prefixed by the association name. The same caveats as for the identifiers for the entities apply here.

> In case of the annotated compositions with the identifier, the change log will contain the entry with identifier's value and will include the change log entries for all annotated elements of the composition's target entity.

Reviewer 2: ```suggestion
If you annotate a composition with an identifier, the change log will contain an entry with the identifier's value. Additionally, it will include change log entries for all annotated elements of the composition's target entity.
```

or

```suggestion
If you annotate a composition with an identifier, the change log will contain an entry with the identifier's value. Additionally, it will include change log entries for all annotated elements of the composition's target entities.
```

?

:::

:::danger Sensitive Data
Do not use an element with [personal data](../guides/data-privacy/annotations#personaldata) as the identifier. Such elements will be ignored.
:::

### Identifiers for Associated Entities

> When your entity has an association to other entity, you might want to log the changes in their relationship.

Reviewer 2: ```suggestion
When your entity has an association to an other entity, you might want to log the changes in their relationship.
```
